base/exporter.go

```diff
@@ -0,0 +1,714 @@
+package base
```

Comment:
Since we're defining all stats here (and not just Prometheus), maybe move the general stats functionality to a stats.go, and potentially stick prometheus specific functionality into stats_prometheus.go?